modules/arrow/src/geoarrow/convert-geoarrow-to-binary-geometry.ts

```diff
@@ -32,33 +36,41 @@ export const BINARY_GEOMETRY_TEMPLATE = {
   featureIds: {value: new Uint32Array(0), size: 1}
 };
 
+export type BinaryGeometriesFromArrowOptions = {
+  // specify which chunk to get binary geometries from, for progressive rendering
+  chunkIndex?: number;
+  // option to get mean centers from geometries, for filtering
+  meanCenter?: boolean;
+};
+
 /**
  * get binary geometries from geoarrow column
  *
  * @param geoColumn the geoarrow column, e.g. arrowTable.getChildAt(geoColumnIndex)
  * @param geoEncoding the geo encoding of the geoarrow column, e.g. getGeoArrowEncoding(arrowTable.schema, geoColumnName)
+ * @param options options for getting binary geometries {meanCenter: boolean}
  * @returns BinaryDataFromGeoArrow
  */
 export function getBinaryGeometriesFromArrow(
   geoColumn: arrow.Vector,
-  geoEncoding: GeoArrowEncoding
+  geoEncoding: GeoArrowEncoding,
+  options?: BinaryGeometriesFromArrowOptions
 ): BinaryDataFromGeoArrow {
   const featureTypes = {
     polygon: geoEncoding === 'geoarrow.multipolygon' || geoEncoding === 'geoarrow.polygon',
     point: geoEncoding === 'geoarrow.multipoint' || geoEncoding === 'geoarrow.point',
     line: geoEncoding === 'geoarrow.multilinestring' || geoEncoding === 'geoarrow.linestring'
   };
 
-  const chunks = geoColumn.data;
+  const chunks = options?.chunkIndex ? [geoColumn.data[options?.chunkIndex]] : geoColumn.data;
   let bounds: [number, number, number, number] = [Infinity, Infinity, -Infinity, -Infinity];
   let globalFeatureIdOffset = 0;
   const binaryGeometries: BinaryFeatures[] = [];
+  const meanCenterArray: Float64Array[] = [];
 
   chunks.forEach((chunk) => {
-    const {featureIds, flatCoordinateArray, nDim, geomOffset} = getBinaryGeometriesFromChunk(
-      chunk,
-      geoEncoding
-    );
+    const {featureIds, flatCoordinateArray, nDim, geomOffset, triangles, meanCenters} =
```

Comment:
That is a lot of return values. And this function is getting really long. Think about simplifying and breaking down.